brightchain-lib/src/lib/services/blockService.ts

```
@@ -0,0 +1,796 @@
+    // For Buffer or SerializableBuffer, require a filePath parameter.
+    if (fileData instanceof Buffer || fileData instanceof SerializableBuffer) {
+      if (filePath) {
+        return basename(filePath);
+      }
+      throw new BlockServiceError(BlockServiceErrorType.FilePathNotProvided);
+    }
+
+    // For Readable streams, use provided filePath or check for a stream 'path' property.
+    const pathToUse =
+      filePath ||
+      ((fileData as any).path &&
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)